### PR TITLE
This will disable FastMuxSwitch if a client is actively transcoding a liveTV stream.

### PR DESCRIPTION
**Pull request by jvl711**
It appears that when SageTV is doing server side transcoding that it is not able to do a fastmuxswitch and have the client transition to the new program/timeslot.  I have noticed this behavior in the Placeshifter, and the andriod client.  With this change, placeshifter is able to transition between timeslots without issue.  And when you are using a client without transcoding, it does a fastmuxswitch without issue.

### CODE REVIEW

File: java/sage/MiniPlayer.java [hunk 1]
```diff
@@ -617,6 +617,20 @@ void checkForByteBasedSeeking(java.io.File file) {
     }
   }
 
+  /***
+   * Determines if this instance of the miniplayer is transcoding
+   * @return Returns true if the miniplayer is transcoding 
+   */
+  public boolean isTranscoding()
+  {
+    if(this.mpegSrc.getTranscoder() != null && serverSideTranscoding)
```
Comment (Narflex) [inline java/sage/MiniPlayer.java:626]: Remove the 'this.' you're using here and other places, it's not needed.

Comment (Narflex) [inline java/sage/MiniPlayer.java:626]: Check if mpegSrc is null before calling a method on it

File: java/sage/Seeker.java [hunk 2]
```diff
@@ -4150,8 +4171,16 @@ else if (currRecMR != null && currRecMR.getSchedulingAiring().getEndTime() <= cu
           {
             if (Sage.DBG) System.out.println("Current record is over.");
             defaultRecord = null;
-            endRecord(es, currTime, switchThis = fastMuxSwitch && es.capDev.supportsFastMuxSwitch() && es.capDev.getRecordedBytes() > 0 &&
-                es.currRecordFile.getSize() > 0);
+
+            switchThis = fastMuxSwitch && es.capDev.supportsFastMuxSwitch() && es.capDev.getRecordedBytes() > 0 && es.currRecordFile.getSize() > 0;
+
+            if(this.isAClientControllingCaptureDevice(es.capDev) && this.isClientTranscodingMediaFile(es.currRecordFile))
```
Comment (Narflex) [inline java/sage/Seeker.java:4177]: Remove the 'this.' and add a space after the 'if'

File: java/sage/Seeker.java [hunk 3]
```diff
@@ -4403,12 +4432,20 @@ else if (currRecords.contains(es.forceWatch) && (es.forceWatch != es.currRecord)
         }
 
         if ((es.currRecord != newRecord) && (es.currRecord != null))
-        {
-          if (Sage.DBG) System.out.println("Change in record, logging recorded data.");
-          endRecord(es, currTime, switchThis = fastMuxSwitch && es.capDev.supportsFastMuxSwitch() && es.capDev.getRecordedBytes() > 0 &&
-              es.currRecordFile.getSize() > 0);
-          switchAny = true;
-        }
+	{
```
Comment (Narflex) [inline java/sage/Seeker.java:4435]: Fix the indentation of this block

File: java/sage/Seeker.java [hunk 4]
```diff
@@ -4403,12 +4432,20 @@ else if (currRecords.contains(es.forceWatch) && (es.forceWatch != es.currRecord)
...
+	  if (Sage.DBG) System.out.println("Change in record, logging recorded data.");
+	  
+	  switchThis = fastMuxSwitch && es.capDev.supportsFastMuxSwitch() && es.capDev.getRecordedBytes() > 0 && es.currRecordFile.getSize() > 0;
+	  
+	  if(this.isAClientControllingCaptureDevice(es.capDev) && this.isClientTranscodingMediaFile(es.currRecordFile))
```
Comment (Narflex) [inline java/sage/Seeker.java:4440]: Same as before

File: java/sage/VideoFrame.java [hunk 5]
```diff
@@ -4829,6 +4829,25 @@ public boolean isCaptureDeviceControlled(CaptureDevice capDev)
...
+  /***
+   * Determine if the client is currently transcoding the mediafile
+   * @param mf The media file object check
+   * @return Returns true if the client is currently transcoding the given mediafile
+   */
+  public boolean isTranscodingMediaFile(MediaFile mf)
+  {    
+    if(player instanceof MiniPlayer)
+    {
+      //((MiniPlayer)player).getFi
```
Comment (Narflex) [inline java/sage/VideoFrame.java:4841]: Remove this line

File: java/sage/VideoFrame.java [hunk 6]
```diff
@@ -4829,6 +4829,25 @@ public boolean isCaptureDeviceControlled(CaptureDevice capDev)
...
+      if(this.getCurrFile() == mf && ((MiniPlayer)player).isTranscoding())
```
Comment (Narflex) [inline java/sage/VideoFrame.java:4842]: Space after if, remove the 'this.'

**Review by Narflex — CHANGES_REQUESTED** [review]

**Comment by jvl711** [comment]
@Narflex 

I think I addressed all of the issues.  Sorry about the multiple pushes.  Some of the formatting issues were only in Git and not showing in my source.  I was taking a while for me to figure it out.

Thanks,
Josh

**Comment by Narflex** [comment]
Thanks!

**Pull request merged**